### ISSUE TITLE
Goals

### ISSUE THREAD

**Issue by brb3** [devanooga/jobs-archive#1]
The goal of this repo is to build a job board for the devanooga group and other Chattanooga-area developers.

Features:
- [ ] Full time job Listings
- [ ] Project-based / contract job listings
- [ ] Integration with the devanooga slack
- [ ] Spam prevention of some kind -- either requiring login to post, a captcha, or some other spam busting (from @7279616e -- Possibly accomplish both by using devanooga slack accounts as auth to the job board? -- [Sign In with Slack](https://api.slack.com/docs/sign-in-with-slack))
- [ ] Source from feeds (RSS? JSON? eg: https://weworkremotely.com/categories/2-programming/jobs.rss)
- [ ] RSS feeds
- [ ] Subscriptions where you can get e-mails for jobs in your chosen fields/languages/salary ranges

**Comment by ryanmaynard** [comment]
> Integration with the devanooga slack

> Spam prevention of some kind -- either requiring login to post, a captcha, or some other spam busting

Possibly accomplish both by using devanooga slack accounts as auth to the job board?
[Sign In with Slack](https://api.slack.com/docs/sign-in-with-slack)

**Comment by brb3** [comment]
Oh, I like the idea of signing in with Slack. I'll tack that onto the original comment.

**Comment by ryanmaynard** [comment]
Please feel free to naysay what I'm about to type. Criticism, and clarification of goals, are highly encouraged. If the goal of this project is like building a birdhouse (Its not the destination, but the journey and the intrinsic reward of DIY), that would disqualify this option.

> The goal of this repo is to **build** a job board for the devanooga group and other Chattanooga-area developers.

So is that a definite "no" to all of the open source options available? Is there something that exists that fits 75% or more of Devanooga's parameters, that we can massage into the desired iteration? If the goal is for the job board to be 'built' by Devanooga, that is more than understandable. However, that is a lot of lost billable hours that could be allocated elsewhere (like recruiting procurement departments to freely post to the forum so it gains traction). 

What about [Jobskee](https://github.com/elinoretenorio/jobskee-open-source-job-board)? Or its hosted version [Tackpost](https://tackpost.com/) as a trial run? [Like so](https://devanooga.tackpost.com/)? <- Allows for multiple admins so that we could explore it all at once.

- [ ] Fulltime/Parttime/Contract could be a tag/category
- [ ] Could integrate with slack by [piping RSS feed](https://slack.com/apps/search?q=rss) into #opportunities
- [ ] Spam-prevention is email login
- [ ] Source from feeds - can bulk upload with csv. wouldn't take much effort to convert rss to csv from select job boards
- [ ] Subscriptions to job categories - I believe its available, but I'm currently overlooking it. 

On the hosted version, Tackpost, its $6/month to remove the branding and to use a custom domain. If we hosted it ourselves, it would run around the same cost on the cheaper end. 

Maybe that is a terrible idea. I don't know; I haven't used it previously. This is just a mild attempt to parse a larger, vague goal into discrete, digestible tasks. 

If any of you want me to send an admin invite to that demo instance, let me know. If not interested, also understandable.

Disclaimer: 

- I haven't worked in DotNet previously, so I likely wouldn't be able to contribute much if you went down that road (not that you would miss out on much ha!).
- I've been in the Sun most of the day, so its likely my processor is fried and I'm overlooking some stupid-obvious obstacle to this notion. There are probably a bunch of dumb grammar/spelling mistakes in this madness.

**Comment by StrangeWill** [comment]
> If the goal is for the job board to be 'built' by Devanooga

Yeah, primary reason was to pick a somewhat useful project to work on, and something that wasn't just a static site. We were _itching_ to find something to do and show off some local dev's skills, not so much that job boards are an unsolved problem.

> However, that is a lot of lost billable hours that could be allocated elsewhere

I'm fine with allocating this somewhere else and pulling a solution off the shelf (though honestly we'd probably just sunset the entire project for now). This was just something thrown up so we got some community project to work on during Hacknights. I'm fine with sponsoring other ideas (it would be nice to have a few on a few different platforms to encourage people on various platforms).

>I haven't worked in DotNet previously, so I likely wouldn't be able to contribute much if you went down that road (not that you would miss out on much ha!).

That's generally the idea though -- a job board is _very_ low hanging fruit and we got some members that were interested in .NET Core (specifically now that it's fully open source and cross platform), so it's pretty easy to get people playing with it.

**Comment by brb3** [comment]
Yea, I'm with both of you guys here, really.

The goal of this project was the _act_ of building something in .NET Core -- not really having a job board. Also, I kinda pushed this idea because I have "noogajobs.com" and think it's a good domain name, lol.

Honestly, I'm kind of _over_ this idea, and would love to have a different project to tackle. I really just want something that we can work on as a team at hacknights. That's the real goal here -- having a "devanooga project" up and running.

Thanks for the input @7279616e. We love having discussions. There are no sacred cows here.

**Comment by ryanmaynard** [comment]
> The goal of this project was the act of building something in .NET Core

Please don't let my last comment kill the project. That may be a good reason for it to be a job board, if the goal is to get some of us unfamiliar with .NET Core to work on something. 

Two ideas, based on the above:

- Continue with the plan to build/hack on a job board with .NET Core. 
-  Another project surfaces for the above goal, and @brb3 uses tackpost or something like it to make use of noogajobs.com? It seems like it would be a fairly low effort thing. Maybe yes? Maybe no? 

In the meantime, I'll ask @StrangeWill for a few ideas/suggestions on a .NET primer so I can begin to prepare for that project.

**Comment by StrangeWill** [comment]
Honestly I felt if another project came up, we'd be unlikely to do that in .NET Core too, unless for some reason the other project is _really_ tasty in .NET Core and no one interested in the other project felt like doing it in another language.

Generally the idea is to let teams that like certain languages fracture out and have projects to work on, as opposed to have one "Devanooga supported" language or even just a handful. _Honestly_ I'd love to see us reach back out to the community and start doing some OSS projects for them at some point, but we're still in our infant stages as far as that is concerned.

---

>In the meantime, I'll ask @StrangeWill for a few ideas/suggestions on a .NET primer so I can begin to prepare for that project.

Other than Microsoft's official docs (https://docs.microsoft.com/en-us/aspnet/core/), don't really have anything of the top of my head to suggest.

--- 

Edit:

Also I don't think we'll ever "can" a project, while @brb3 may not be interested in working on it, I got some people that are. As long as software doesn't become a _liability_ (eg: a security-ridden piece of trash), we'll keep projects up and I'll help maintain them.

**Comment by brb3** [comment]
Yea, this repo isn't going anywhere, and I really want someone to run with this -- don't get me wrong.

However, I don't feel like there's enough traction in the community for this to be our first _big_ community project.

**Comment by StrangeWill** [comment]
I'm pretty sure our first "big" project will be Chattanooga community involved (if we ever get one, it was hard to get traction doing so before too).

**Comment by ryanmaynard** [comment]
Thinking aloud:

Perhaps one that has some non-programmer tasks available, or very simple development tasks, to maximize inclusion? Like your idea behind the job board in .Net Core?

**Comment by StrangeWill** [comment]
We have a lot of meta-Devanooga stuff to be done, nothing glorious, I need to push what markdown files I have so we can point people in that direction.

On top of that, I'd _love_ to have some people reach out to other places in the community (for projects, donations, new members, etc), though I think it'll be a little early for that (At least need our 501(c)3 to come through before then IMO).

**Comment by ryanmaynard** [comment]
No rush on pushing that Markdown file now, especially if you're still feeling sick, sir. I wonder if there is a Slack integration or something that would allow a channel for a makeshift checklist? The idea being that Slack would be a bit more reachable for those newer/uncomfortable with git if the tasks on the list are meta. 

I'll be happy to champion outreach when you give the green light.

**Comment by StrangeWill** [comment]
Also just mentioning it here because it was mentioned in Slack: one of the low-bars-to-entry _is_ to write on our blog (that maybe we'll have someday): https://github.com/devanooga/devanooga.com/issues/43
